core/src/main/java/com/linecorp/armeria/server/file/MimeTypeFunction.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.file;
+
+import static java.util.Objects.requireNonNull;
+
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+/**
+ * A function used for determining the {@link MediaType} of a file based on its path.
+ */
+@UnstableApi
+public interface MimeTypeFunction {
```

Comment:
How about renaming this to `MediaTypeResolver`? I'm suggesting a rename because:
- 'function' sounds somewhat vague; and
- it actually returns a `MediaType` rather than a `MimeType`.
When renaming, please make sure all other fields, methods and parameters are renamed. :bow:

Reply:
I agreed. 👌

Reply:
@trustin Should I rename MimeTypeUtil to MediaTypeUtil as well?

Reply:
Yes please :bow: